language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/CompletionUtil.java

```diff
@@ -88,6 +89,7 @@ private CompletionUtil() {
      */
     public static List<LSCompletionItem> route(BallerinaCompletionContext ctx, Node node)
             throws LSCompletionException {
+        ctx.checkCancelled();
```

Comment:
Shall we add the check in the `fillTokenInfoAtCusror` method as well. As we are going to traverse the syntax tree there

Reply:
The `fillTokenInfoAtCusror` method internally uses the heavy operations such as retrieving the current document, which are being checked for the cancellation within the respective methods (`context.currentDocument`)